lib/core/src/sdk.rs

```diff
@@ -456,20 +457,92 @@ impl LiquidSdk {
+            Ok(RevSwapStates::TransactionMempool) => {
+                let Update::TransactionMempool {
+                    id, transaction, ..
+                } = update
+                else {
+                    return Err(anyhow!("Unexpected payload from Boltz status stream"));
+                };
+
+                let lockup_tx = utils::deserialize_tx_hex(&transaction.hex)?;
+
+                // If the amount is greater than the zero-conf limit
+                // TODO Make limit user-definable through config
+                let receiver_amount_sat = receive_swap.receiver_amount_sat;
+                if receiver_amount_sat > LIQUID_SDK_ZERO_CONF_LIMIT_SAT {
+                    debug!("[Receive Swap {id}] Amount is too high to claim with zero-conf ({receiver_amount_sat} sat > {LIQUID_SDK_ZERO_CONF_LIMIT_SAT} sat). Waiting for confirmation...");
+                    return Ok(());
+                }
+
+                debug!("[Receive Swap {id}] Amount is within valid range for zero-conf ({receiver_amount_sat} < {LIQUID_SDK_ZERO_CONF_LIMIT_SAT} sat)");
+
+                // If the transaction has RBF, see https://github.com/bitcoin/bips/blob/master/bip-0125.mediawiki
+                let rbf_explicit = lockup_tx.input.iter().any(|input| input.sequence.is_rbf());
+                // let rbf_implicit = lockup_tx_history.height < 0;
+
+                if rbf_explicit {
+                    debug!("[Receive Swap {id}] Lockup transaction signals RBF. Waiting for confirmation...");
+                    return Ok(());
+                }
+
+                debug!("[Receive Swap {id}] Lockup tx does not signal RBF. Proceeding...");
+
+                // If the fees are higher than our estimated value
+                let tx_fees: u64 = lockup_tx.all_fees().values().sum();
+                let min_fee_rate = match self.config.network {
+                    Network::Mainnet => LIQUID_SDK_ZERO_CONF_FEE_RATE_MAINNET,
+                    Network::Testnet => LIQUID_SDK_ZERO_CONF_FEE_RATE_TESTNET,
+                };
+                let lower_bound_estimated_fees = lockup_tx.vsize() as f32 * min_fee_rate * 0.8;
+
+                if lower_bound_estimated_fees > tx_fees as f32 {
+                    debug!("[Receive Swap {id}] Lockup tx fees are too low: Expected at least {lower_bound_estimated_fees} sat, got {tx_fees} sat. Waiting for confirmation...");
+                    return Ok(());
+                }
+
+                debug!("[Receive Swap {id}] Lockup tx fees are within acceptable range ({tx_fees} > {lower_bound_estimated_fees} sat). Proceeding with claim.");
+
+                match receive_swap.claim_tx_id {
+                    Some(claim_tx_id) => {
+                        warn!("Claim tx for Receive Swap {id} was already broadcast: txid {claim_tx_id}")
+                    }
+                    None => {
+                        self.try_handle_receive_swap_update(&receive_swap.id, Pending, None)
```

Comment:
Should this state transition and event be triggered before our 0-conf preconditions?

In other words, as soon as we see the lockup tx in the mempool?

Reply:
The switch to pending state has been removed and handled in a separate PR (#301)